tests/lax_test.py

```diff
@@ -1161,6 +1161,41 @@ def testDotGeneralAgainstNumpy(self, lhs_shape, rhs_shape, dtype,
     numpy_op = lambda x, y: lax_reference.dot_general(x, y, dimension_numbers)
     self._CheckAgainstNumpy(numpy_op, op, args_maker)
 
+  @jtu.sample_product(
+      [
+          dict(
+              m=m, k=k, n=n, num_groups=num_groups,
+          )
+          for m, k, n, num_groups in [
+              (5, 4, 3, 1),
+              (10, 9, 8, 2),
+          ]
+      ],
+      dtype=lax_test_util.all_dtypes,
+  )
+  def testRaggedDot(self, m, k, n, num_groups, dtype):
+    """Tests ragged_dot.
+
+    The ragged_dot is tested against numpy reference implementation, and by running JAX compilation.
+
+    Raises:
+      SkipTest: in the case dtype is not supported.
+    """
+    if np.issubdtype(dtype, np.bool_):
+      # This is due to "reduce_sum does not accept dtype bool".
+      raise SkipTest('ragged_dot for bool not supported')
```

Comment:
Rather than skipping bool in the implementation, you should skip bool in the parameterization, e.g use `jtu.dtypes.numeric`. Since only `N` test cases are generated per test, this leads to more test coverage.

Reply:
Great feedback, done!